org.eclipse.jface.text/src/org/eclipse/jface/text/source/SourceViewer.java

```diff
@@ -428,7 +428,7 @@ public SourceViewer(Composite parent, IVerticalRuler verticalRuler, IOverviewRul
 		fIsVerticalRulerVisible= (verticalRuler != null);
 		fOverviewRuler= overviewRuler;
 		fIsOverviewRulerVisible= (showAnnotationsOverview && overviewRuler != null);
-		this.lifecycles= new HashSet<>();
+		this.lifecycles= new ArrayList<>();
```

Comment:
Iff you install / uninstall in the *same* order, a LinkedHashSet might be easier.

I'd assume that uninstall should happen in the opposite order as the install, no?

Reply:
in the old version install and remove order was the same

Reply:
Thank you for checking